util.py

```diff
@@ -250,8 +251,16 @@ def fetchFile(source, dest):
                source[:6] == 'https:' or \
                source[:5] == 'file:' or \
                source[:4] == 'ftp:':
-            # This something that can be fetched using urllib2:
-            fd = urllib2.urlopen(source)
+            # This something that can be fetched using urllib2
+            request = source
+
+            if source.startswith('http') and '@' in source: # Auth HTTP(s)
```

Comment:
`@` is a valid symbol to have in a URL path so this check is overzealous. Rather construct the URL object first and then use `getScheme()`/`getUsername()`/`getPassword()` to check if HTTP basic authentication needs to be set up.